install.ps1

```diff
@@ -0,0 +1,132 @@
+#>
+[cmdletbinding()]
+param(
+    [string]$InstallDir="<auto>"
+)
+
+function Get-Install-Directory([string]$InstallDir) {
+    $dir = "OpenTelemetry .NET AutoInstrumentation"
+    
+    if($InstallDir -eq "<auto>" -or $installDir -eq "AppData") {
+        return (Join-Path $env:LOCALAPPDATA "Programs" | Join-Path -ChildPath $dir)
+    }
+    elseif($InstallDir -eq "ProgramFiles"){
+        return (Join-Path "C:\Program Files\" $dir)
+    } 
+    elseif(Test-Path $InstallDir -IsValid) {
+        return $InstallDir
+    }
+
+    throw "Invalid install directory provided '$InstallDir'"
+}
+
+function Get-Temp-Directory() {
+    return $env:temp
+}
+
+function Prepare-Install-Directory([string]$InstallDir) {
+    New-Item -ItemType Directory -Force -Path $InstallDir | Out-Null
+}
+
+function Download-OpenTelemetry([string]$Version, [string]$Path) {
+    $archive = "opentelemetry-dotnet-instrumentation-windows.zip"
+    $dlUrl = "https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/releases/download/$Version/$archive"
+    $dlPath = Join-Path $Path $archive
+
+    Invoke-WebRequest -Uri $dlUrl -OutFile $dlPath
+
+    return $dlPath
+}
+
+function Extract-OpenTelemetry([string]$DlPath, [string] $InstallPath) {
+    Expand-Archive $DlPath $InstallPath -Force
+}
+
+function Setup-OpenTelemetry-Environment([string] $InstallPath) {
+    $target = [System.EnvironmentVariableTarget]::Machine
+
+    # .NET Framework
+    [System.Environment]::SetEnvironmentVariable('COR_PROFILER','{918728DD-259F-4A6A-AC2B-B85E1B658318}', $target)
+    [System.Environment]::SetEnvironmentVariable('COR_PROFILER_PATH_32', (Join-Path $InstallPath "/win-x86/OpenTelemetry.AutoInstrumentation.Native.dll"), $target)
+    [System.Environment]::SetEnvironmentVariable('COR_PROFILER_PATH_64', (Join-Path $InstallPath "/win-x64/OpenTelemetry.AutoInstrumentation.Native.dll"), $target)
+
+    # .NET Core
+    [System.Environment]::SetEnvironmentVariable('CORECLR_PROFILER','{918728DD-259F-4A6A-AC2B-B85E1B658318}', $target)
+    [System.Environment]::SetEnvironmentVariable('CORECLR_PROFILER_PATH_32', (Join-Path $InstallPath "/win-x86/OpenTelemetry.AutoInstrumentation.Native.dll"), $target)
+    [System.Environment]::SetEnvironmentVariable('CORECLR_PROFILER_PATH_64', (Join-Path $InstallPath "/win-x64/OpenTelemetry.AutoInstrumentation.Native.dll"), $target)
+
+    # .NET Common
+    [System.Environment]::SetEnvironmentVariable('DOTNET_ADDITIONAL_DEPS', (Join-Path $InstallPath "AdditionalDeps"), $target)
+    [System.Environment]::SetEnvironmentVariable('DOTNET_SHARED_STORE', (Join-Path $InstallPath "store"), $target)
+    [System.Environment]::SetEnvironmentVariable('DOTNET_STARTUP_HOOKS', (Join-Path $InstallPath "netcoreapp3.1/OpenTelemetry.AutoInstrumentation.StartupHook.dll"), $target)
+
+    # ASP.NET Core
+    [System.Environment]::SetEnvironmentVariable('ASPNETCORE_HOSTINGSTARTUPASSEMBLIES', "OpenTelemetry.AutoInstrumentation.AspNetCoreBootstrapper", $target)
```

Comment:
1. It will instrument **all** the .NET (Core) applications (without bytecode instrumentation). Is it not a problem?
2. Take notice that the instrumented application may already use `DOTNET_STARTUP_HOOKS` (or other .NET runtime env var). Then the auto-instrumentation will not work (the same applies to `./instrument.sh` I will create an issue for that).

Reply:
ad. 2 for reference: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1390

Reply:
`ASPNETCORE_HOSTINGSTARTUPASSEMBLIES` can be moved to IIS scope, but then it will also break ASP.NET Core + other than IIS webhost. Since it's not a path-based variable, but assembly lookup by the name, then nothing should happen, if it can't find that assembly loaded. Unless it's present in runtime store? (but we shouldn't put our libraries there?)

Reply:
Point number 1 is also about `DOTNET_ADDITIONAL_DEPS`, `DOTNET_SHARED_STORE`, `DOTNET_STARTUP_HOOKS`

Reply:
Great, forgot about those. You are right.